implementations/rust/ockam/ockam_node/src/storage/database/migrations/20231006100000_create_database.sql

```diff
@@ -0,0 +1,187 @@
+CREATE TABLE identity
+(
+    identifier     TEXT NOT NULL UNIQUE,
```

Comment:
I think that we should start favoring debuggability unless there are established performance concerns. Having the possibility to read text values compared to having blobs is really useful when you're diagnosing an incident or a failed test.

Reply:
> That's true, but you can always print it as a string. SQL GUIs usually show it as a hex anyways
> I suspect you can convert it to hex inside the SQL SELECT statement if you wish

When you are debugging complicated things, you generally don't want to write additional code or commands to compare for example the content of the database to a line in a log file. Unless there is a strong need to do otherwise I think we get lots of value to be able to see all the data directly as much as we can.

Reply:
Could you please elaborate? Maybe with a small example? Because in my opinion, we should pick the most suitable way of storing data. In this case we have binary data and SQL supports binary fields. I don't understand how picking the more compact format for storing makes querying/displaying/logging data and debugging application harder.

Reply:
For example a binary blob in TablePlus (SQL client for MacOS) shows up like this
<img width="320" alt="image" src="https://github.com/build-trust/ockam/assets/10988/b7dd68da-8d34-4bd8-8a28-b1a8319e6ad6">
If identifiers are stored as blob it will be hard to see what they are, especially if we want to correlate with logs.

Reply:
You made me download the app :D
![Uploading Screenshot 2023-11-21 at 21.04.58.png…]()

Reply:
Printing it as a hex to the log is easy, selecting it from the database in hex format is also easy

Reply:
I agree, but that complain should likely be directed to the creator of TablePlus. Both GUIs I use (Intellij IDEs' embedded sql tool and DBeaver) have no problem displaying blobs as hex without any extra clicks.

Reply:
That's good I didn't know that some GUIs could be configured for this. However:

 - this is not the case for all of them, for example DB Browser doesn't do it as far as I can see
 - if we need to ssh somewhere and the only thing we have to diagnose an issue is the command-line `sqlite`, we're stuck